woocommerce/woocommerce-ios
# Shipping Labels: Fix Back/Done buttons on Add New Package screen if store has no existing packages

Fixes: #4937
⚠️ Don't merge before #4916 ⚠️ 

## Description

In the Shipping Label flow, when creating a new package the back and Done buttons on the Add New Package screen didn't have consistent behavior:

1. If your store has no packages set up yet, tapping the back and Done buttons doesn't dismiss the view. (It should dismiss the modal so you're taken back to the Package Details screen.)
2. If your store does have packages set up, tapping the back and Done buttons dismisses the view. (This navigates you back to the Package Selected modal view.)

This PR fixes those buttons so they consistently dismiss the view as expected.

## Changes

Previously, the navigation buttons were defined on the Add New Package child views, because the Done buttons have different behavior depending on what tab you have selected (Custom Package or Service Package). However, when the Add New Package screen is the first screen displayed in the modal sheet, and the environment's presentation mode binding property `presentation` is set in the child view, the presentation mode's wrapped value `isPresented` is `false` (instead of `true`) so the parent view isn't dismissed with `presentation.wrappedValue.dismiss()`. The same issue happens when using the `minimalNavigationBarBackButton()` view modifier from within the body of `ShippingLabelAddNewPackage`.

To resolve this:

* The Done buttons are moved from the child views to a single button in `ShippingLabelAddNewPackage`. (The specific actions that button, depending on the view, still need to be completed in an upcoming PR.)
* The back button is moved from the child views to `ShippingLabelAddNewPackage`, without using the `minimalNavigationBarBackButton()` view modifier (since the behavior isn't consistent when used in this context).

<img src="https://user-images.githubusercontent.com/8658164/132225266-6de808a3-7927-4b16-a7c1-ab3101c8f6b2.gif" width="300px">

## Testing

1. Make sure you have configured the WooCommerce WCShip plugin, and there are no packages set up in WP Admin under WooCommerce > Settings > Shipping > WooCommerce Shipping.
2. Launch the app in debug mode (so the feature flag is enabled).
3. Open an order detail that is eligible for the shipping label.
4. Tap the button "Create Shipping Label.”
5. Confirm "Ship From" and "Ship To.”
6. In the package details section, tap on “Package Selected.”
7. On the Add New Package screen, tap the back button and confirm it dismisses the view.
8. Reopen the Add New Package screen and tap the Done button on each tab, confirming it dismisses the view. (On the Custom Package tab, you have to fill out the form to resolve any validation errors and then tap Done.)

You can also repeat these steps with packages set up in WP Admin under WooCommerce > Settings > Shipping > WooCommerce Shipping, and confirm the buttons continue to work as expected.

## Submitter Checklist

Update release notes:
